spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -3463,15 +3477,14 @@ public void nack(int index, Duration sleep) {
 				for (ConsumerRecord<K, V> cRecord : this.records) {
 					if (i++ < index) {
 						toAck.add(cRecord);
-					}
-					else {
+					} else {
 						break;
 					}
 				}
 				Map<TopicPartition, List<ConsumerRecord<K, V>>> newRecords = new HashMap<>();
 				for (ConsumerRecord<K, V> cRecord : toAck) {
 					newRecords.computeIfAbsent(new TopicPartition(cRecord.topic(), cRecord.partition()),
-							tp -> new LinkedList<>()).add(cRecord);
+											   tp -> new LinkedList<>()).add(cRecord);
```

Comment:
I think if we get rid of that `Acknowledgment.isAnyManualAck()` all the refactoring changes in the class are void.

Reply:
Thank you for confirming that we are on the same page about `Acknowledgment`.
Would you mind to revert changes in this class since they are just code style refactoring and apparently not the one we are following. Especially that `} else {` block.

Reply:
No, it's not true. 
If `ack != null` is true, it's equivalent to `anyManualAck`, so the refactoring is still valid.

Previously, even if the filter resulted in an empty list, the listener was always called. With this PR, the listener is now only called when a Manual Ack is required. since consumerAware was always true, the listener was called even in cases that did not require manual Ack. In this case, regardless of `consumerAware == true`, if it is not a manual Ack and the list is empty, the listener will not be called.

```java
// FilteringBatchMessageListenerAdapater#onMessage() 
...
if (consumerRecords.isEmpty()) {
			if (acknowledgment != null) {
				invokeDelegate(consumerRecords, acknowledgment, consumer);
			}
		}
		else {
			if (this.consumerAware
				|| (!this.ackDiscarded && this.delegateType.equals(ListenerType.ACKNOWLEDGING))) {
				invokeDelegate(consumerRecords, acknowledgment, consumer);
			}
			else {
				if (this.ackDiscarded && acknowledgment != null) {
					acknowledgment.acknowledge();
				}
			}
		}
...
```